execute/executor.go

```diff
@@ -275,8 +302,8 @@ func (ec executionContext) Allocator() *Allocator {
 func (ec executionContext) Parents() []DatasetID {
 	return ec.parents
 }
-func (ec executionContext) ConvertID(id planner.ProcedureID) DatasetID {
-	return DatasetID(plan.ProcedureID(id))
+func (ec executionContext) ConvertID(id plan.ProcedureID) DatasetID {
```

Comment:
So I guess we still need procedure ids?  Seems like there should be a way to do away with them.